compiler/guards.nim

```diff
@@ -981,7 +981,7 @@ proc buildElse(n: PNode; o: Operators): PNode =
   var s = newNodeIT(nkCurly, n.info, settype(n.sons[0]))
   for i in 1..n.len-2:
     let branch = n.sons[i]
-    assert branch.kind == nkOfBranch
+    assert branch.kind == nkElse
```

Comment:
No, the logic really assumes ``nkOfBranch`` here. There is something else going on.